src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,153 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
+import {
+  getPostDetails,
+  getPostComments,
+  removeComment,
+  addComment,
+} from '../../api';
 import './PostDetails.scss';
+import { Loader } from '../Loader';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
+interface Props {
+  selectedPostId: number;
+}
+
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [post, setPost] = useState<Post | null>(null);
+  const [isPostLoading, setIsPostLoading] = useState(false);
+  const [comments, setComments] = useState<PostComment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [isCommentsShow, toggleIsCommentsShow] = useState(true);
+
+  const loadPostById = async (postId: number) => {
+    try {
+      setIsPostLoading(true);
+      const postsFromServer = await getPostDetails(postId);
+
+      setPost(postsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsPostLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPostById(selectedPostId);
+  }, [selectedPostId]);
+
+  const loadCommentsByPostId = async (postId: number) => {
+    try {
+      setIsCommentsLoading(true);
+      const commentsFromServer = await getPostComments(postId);
+
+      setComments(commentsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsCommentsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadCommentsByPostId(selectedPostId);
+  }, [selectedPostId]);
```

Comment:
Combine your 2 `useEffect` into one, and use `Promise.all` to load data parallel